src/Rest/RestApiBrowser.php

```diff
@@ -27,23 +25,20 @@ class RestApiBrowser
     /** @var ResponseStorage */
     private $responseStorage;
 
+    /** @var string */
+    private $host;
+
+    /** @var MessageFactoryDiscovery */
+    private $messageFactory;
+
     /**
-     * @param string $base_url
-     * @param string|null $adaptor_name
-     * @throws HttpAdapterException
+     * @param string $host
      */
-    public function __construct($base_url, $adaptor_name, HttpClient $httpClient = null)
+    public function __construct($host, HttpClient $httpClient = null)
     {
-        if (!is_null($httpClient) && $httpClient instanceof HttpClient) {
-            $this->httpClient = $httpClient;
-        } else {
-            if (is_string($adaptor_name) && HttpAdapterFactory::capable($adaptor_name)) {
-                $this->httpClient = HttpAdapterFactory::create($adaptor_name);
-            } else {
-                $this->httpClient = HttpAdapterFactory::guess();
-            }
-            $this->httpClient->getConfiguration()->setBaseUri($base_url);
-        }
+        $this->host = $host;
+        $this->httpClient = $httpClient ?: HttpClientDiscovery::find();
```

Comment:
What happen if we don't have required any http client in our composer.json? Some exception is throwed?